beater/beater.go

```diff
@@ -21,18 +22,15 @@ type beater struct {
 func New(b *beat.Beat, ucfg *common.Config) (beat.Beater, error) {
 	beaterConfig := defaultConfig()
 	if err := ucfg.Unpack(beaterConfig); err != nil {
-		return nil, fmt.Errorf("Error reading config file: %v", err)
+		return nil, fmt.Errorf("error reading config file: %v", err)
+	}
+
+	if err := validateConfig(beaterConfig, b.Config); err != nil {
+		return nil, err
 	}
+
 	if beaterConfig.Frontend.isEnabled() {
-		if _, err := regexp.Compile(beaterConfig.Frontend.LibraryPattern); err != nil {
-			return nil, errors.New(fmt.Sprintf("Invalid regex for `library_pattern`: %v", err.Error()))
-		}
-		if _, err := regexp.Compile(beaterConfig.Frontend.ExcludeFromGrouping); err != nil {
-			return nil, errors.New(fmt.Sprintf("Invalid regex for `exclude_from_grouping`: %v", err.Error()))
-		}
-		if b.Config != nil && b.Config.Output.Name() == "elasticsearch" {
-			beaterConfig.setElasticsearch(b.Config.Output.Config())
-		}
+		beaterConfig.setElasticsearch(b.Config.Output.Config())
```

Comment:
why did you remove th check for `b.Config != nil && b.Config.Output.Name() == "elasticsearch"` here?